File: app/src/main/java/pl/mosenko/sunnypodlaskie/mvp/weatherdatalist/WeatherDataListContract.java
package pl.mosenko.sunnypodlaskie.mvp.weatherdatalist;

import com.hannesdorfmann.mosby3.mvp.MvpPresenter;
import com.hannesdorfmann.mosby3.mvp.lce.MvpLceView;

import java.util.List;

import pl.mosenko.sunnypodlaskie.persistence.entities.WeatherDataEntity;

/**
 * Created by syk on 06.06.17.
 */

public interface WeatherDataListContract {

    interface View extends MvpLceView<List<WeatherDataEntity>> {
        void showEmpty();

        void showError(Throwable throwable);

        void showDataWithoutInternetUpdatedMessage();
    }

    interface Presenter extends MvpPresenter<View> {
        void loadData(boolean pullToRefresh);

        void onResume();

        void onPause();
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/mvp/weatherdatalist/WeatherDataListContract.kt
package pl.mosenko.sunnypodlaskie.mvp.weatherdatalist

import pl.mosenko.sunnypodlaskie.mvp.MvpPresenter
import pl.mosenko.sunnypodlaskie.mvp.MvpView
import pl.mosenko.sunnypodlaskie.persistence.entities.WeatherDataEntity

/**
 * Created by syk on 06.06.17.
 */
interface WeatherDataListContract {
    interface View : MvpView {
        fun showLoading(pullToRefresh: Boolean)
        fun showContent()
        fun showError(e: Throwable, pullToRefresh: Boolean)
        fun setData(data: MutableList<WeatherDataEntity>)
        fun loadData(pullToRefresh: Boolean)
        fun showEmpty()
        fun showError(throwable: Throwable)
        fun showDataWithoutInternetUpdatedMessage()
    }

    interface Presenter : MvpPresenter<View> {
        fun loadData(pullToRefresh: Boolean)
        fun onResume()
        fun onPause()
    }
}
